lnwire/onion_msg_payload.go

```diff
@@ -0,0 +1,428 @@
+package lnwire
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"sort"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/lightningnetwork/lnd/tlv"
+)
+
+const (
+	// finalHopPayloadStart is the inclusive beginning of the tlv type
+	// range that is reserved for payloads for the final hop.
+	finalHopPayloadStart tlv.Type = 64
+
+	// replyPathType is a record for onion messaging reply paths.
+	replyPathType tlv.Type = 2
+
+	// encryptedDataTLVType is a record containing encrypted data for
+	// message recipient.
+	encryptedDataTLVType tlv.Type = 4
+
+	// InvoiceRequestNamespaceType is a record containing the sub-namespace
+	// of tlvs that request invoices for offers.
+	InvoiceRequestNamespaceType tlv.Type = 64
+
+	// InvoiceNamespaceType is a record containing the sub-namespace of
+	// tlvs that describe an invoice.
+	InvoiceNamespaceType tlv.Type = 66
+
+	// InvoiceErrorNamespaceType is a record containing the sub-namespace of
+	// tlvs that describe an invoice error.
+	InvoiceErrorNamespaceType tlv.Type = 68
+)
+
+var (
+	// ErrNotFinalPayload is returned when a final hop payload is not
+	// within the correct range.
+	ErrNotFinalPayload = errors.New("final hop payloads type should be " +
+		">= 64")
+
+	// ErrNoHops is returned when we handle a reply path that does not
+	// have any hops (this makes no sense).
+	ErrNoHops = errors.New("reply path requires hops")
+)
+
+// OnionMessagePayload contains the contents of an onion message payload.
+type OnionMessagePayload struct {
+	// ReplyPath contains a blinded path that can be used to respond to an
+	// onion message.
+	ReplyPath *ReplyPath
+
+	// EncryptedData contains encrypted data for the recipient.
+	EncryptedData []byte
+
+	// FinalHopPayloads contains any tlvs with type > 64 that
```

Comment:
Proposal: `FinalHopTLVs []*FinalHopTLV`?

I kinda find it white confusing below to have payload in payloads ?